### PR TITLE
Add retry for opensearch client creation in ISM Policy reconciler

### PR DESCRIPTION
**Pull request by nilushancosta**
### Description
Add retry for opensearch client creation in ISM policy reconciler to fix panic
Minor change - Remove extra whitespace in developing.md file

### Issues Resolved
Resolves https://github.com/opensearch-project/opensearch-k8s-operator/issues/801

### Check List
- [x] Commits are signed per the DCO using --signoff 
- [ ] Unittest added for the new/changed functionality and all unit tests are successful
- [ ] Customer-visible features documented
- [x] No linter warnings (`make lint`)

If CRDs are changed:
- [ ] CRD YAMLs updated (`make manifests`) and also copied into the helm chart
- [ ] Changes to CRDs documented

Please refer to the [PR guidelines](https://github.com/opensearch-project/opensearch-k8s-operator/blob/main/docs/developing.md#submitting-a-pr) before submitting this pull request.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by nilushancosta** [comment]
As part of this fix, I modified 2 files; `opensearch-operator/pkg/reconcilers/ismpolicy.go` and `docs/developing.md`. However when I executed `make build manifests`, other files also got modified automatically. Since these modifications included changes to the CRD YAMLs, I copied them to the Helm chart as per the instructions in the developer guide. Please let me know if what I did is correct

File: opensearch-operator/pkg/reconcilers/ismpolicy.go [hunk 1]
```diff
@@ -146,6 +146,7 @@ func (r *IsmPolicyReconciler) Reconcile() (retResult ctrl.Result, retErr error)
 	if err != nil {
 		reason := "error creating opensearch client"
 		r.recorder.Event(r.instance, "Warning", opensearchError, reason)
+		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, err
```
Comment (swoehrl-mw) [inline opensearch-operator/pkg/reconcilers/ismpolicy.go:149]: The correct way here is to set `retResult` and `retErr` as the defer logic will use that to set a corresponding status in kubernetes. Similar to how the blocks above do it.

**Review by swoehrl-mw — CHANGES_REQUESTED** [review]
Hi @nilushancosta. Thanks for contributing this fix.

Not sure why there were changes in the CRD yamls, these should have already been done in the PR where the corresponding code change was introduced. No idea how that slipped through.
In the end, not your fault or problem, you did the copying correctly.

**Comment by swoehrl-mw** [comment]
@nilushancosta On second thought, please remove all the crd yaml changes from the PR (from the chart and the config/crd/bases folder), we'll deal with that separately since the wrong field name (`timestamp_field` vs `timestampField`) in the YAMLs is already part of the 2.6.0 release so cannot be changed without causing inconsistencies.

cc @prudhvigodithi @salyh

File: opensearch-operator/pkg/reconcilers/ismpolicy.go (hunk 1 above)
Comment (nilushancosta) [reply opensearch-operator/pkg/reconcilers/ismpolicy.go]: Thanks for the feedback. I updated the PR

**Comment by nilushancosta** [comment]
> @nilushancosta On second thought, please remove all the crd yaml changes from the PR (from the chart and the config/crd/bases folder), we'll deal with that separately since the wrong field name (`timestamp_field` vs `timestampField`) in the YAMLs is already part of the 2.6.0 release so cannot be changed without causing inconsistencies.
> 
> cc @prudhvigodithi @salyh

Updated the PR by removing them

**Review by swoehrl-mw — APPROVED** [review]

**Pull request merged**